src/NumericalIntegration.cpp

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright 2012-2019 CNRS-UM LIRMM, CNRS-AIST JRL
+ */
+
+// associated header
+#include "RBDyn/NumericalIntegration.h"
+
+// includes
+// RBDyn
+#include "RBDyn/MultiBody.h"
+#include "RBDyn/MultiBodyConfig.h"
+
+using namespace Eigen;
+
+namespace
+{
+
+/** Compute the sum of the first terms of the Magnus expansion of \Omega such that
+ * q' = q*exp(\Omega) is the quaternion obtained after applying a constant acceleration
+ * rotation wD for a duration step, while starting with a velocity w
+ * The function computes the i first terms of the sum, such that ||O_i+1|| < absEps and
+ * ||O_i+1|| < relEps * ||O1||. It stops at the 5th term even if the conditions where not
+ * met, in which case the return bool is set to true as a warning.
+ */
+std::pair<Vector3d, bool> magnusExpansion(const Vector3d & w, const Vector3d & wD, double step,
```

Comment:
I think the return type is not intuitive, the function returns `false` if it worked and `true` if we don't know.
At the 5th step, there is no check whether or not the operation succeed and the return value is always `true` but could also be a `false`.